include/navigation/perfect_memory_store_raw.h

```diff
@@ -71,7 +69,8 @@ class RawImage
     {
         // Calculate difference between image and stored image
         const int imSize = image.rows * image.cols;
-        auto diffIter = m_Differencer(image, m_Snapshots[snapshot], m_DiffScratchImage);
+        thread_local cv::Mat diffScratchImage(image.size(), image.type());
```

Comment:
sadly, it seems (https://stackoverflow.com/a/21376829/1476754) that mixing C++11 ``thread_local`` and OpenMP isn't generally a good idea. Maybe you can use the ``threadprivate`` pragma instead?

Reply:
The reason I did this initially was because I didn't know that you could use the ``#pragma omp`` to do this in random places in your code away from the loop... but I just tried it and it seems to work! You're right that this seems a more sensible approach.